Code-the-Dream-School/ee-prac-team4-front
# set login expiration in local storage

_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._

## Description
This PR adds login expiration to local storage on login or registration. IsLoggedIn state was not persisting across renders because we weren't checking (and can't) if a user has an active token in cookies. This value is set to expire at the same time the token does. 



## Related Issue



## Acceptance Criteria



## Type of Changes



|     | Type                       |
| --- | -------------------------- |
| ✓ | :bug: Bug fix              |
|   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Updates

### Before



### After



## Testing Steps / QA Criteria
1. pull this branch down to your local machine and run npm start as usual (don't forget to run the BE server)
2. navigate to localhost:3000 and open the application tab in the browser
3. register a new user
4. check that an expiry is saved to local storage and that isLoggedIn state is true
5. refresh the app, check that there were no changes to expiry and isLoggedIn state
6. logout and check that expiry was removed from local storage and isLoggedIn is false
7. login to the app and check that expiry is set in local storage again and that isLoggedIn is true



© 2022 GitHub, Inc.
Terms
Privacy
Security
Status
